README.md

```diff
@@ -15,28 +15,129 @@ This is a promise based interface for interacting with a source control manageme
 ### configure
 The `configure` function takes in an object and resets the configuration values
 
-### formatScmUrl
+### parseUrl
 Required parameters:
 
 | Parameter        | Type  |  Description |
 | :-------------   | :---- | :-------------|
-| scmUrl        | String | Scm Url to format |
+| config             | Object | Configuration Object |
+| config.checkoutUrl | String | Checkout url for a repo to parse |
+| config.token  | String | The github token to check permissions on |
 
 #### Expected Outcome
-A formatted scm url to be used as a unique key. This function is synchronous.
+An scmUri (ex: `github.com:1234:branchName`, where 1234 is a repo ID number), which will be a unique identifier for the repo and branch in Screwdriver.
+
+#### Expected Promise response
+1. Resolve with an scm uri for the repository
+2. Reject if not able to parse url
+
+### parseHook
+Required parameters:
+
+| Parameter        | Type  |  Description |
+| :-------------   | :---- | :-------------|
+| headers        | Object | Request header |
+| payload        | Object | Request payload |
+
+#### Expected Outcome
+An object with the following fields:
+```js
+{
+    type: 'pr',         // or 'repo'
+    action: 'opened',   // or 'closed' for type 'pr'; 'push' for type 'repo'
+    username: 'batman',
+    url: 'https://bitbucket.org/batman/test.git',
+    branch: 'mynewbranch',
+    sha: '40171b678527',
+    prNumber: 3,
+    prRef: 'refs/pull-requests/3/from'
+}
+```
+
+### decorateUrl
+Required parameters:
+
+| Parameter        | Type  |  Description |
+| :-------------   | :---- | :-------------|
+| config        | Object | Configuration Object |
+| config.scmUri | String | Scm uri (ex: `github.com:1234:branchName`) |
+| config.token  | String | The github token to check permissions on |
+
+#### Expected Outcome
+Decorated url in the form of:
+```js
+{
+    url: 'https://github.com/screwdriver-cd/scm-base',
+    title: 'screwdriver-cd/scm-base',
+    subtitle: 'branchName'
+}
+```
+
+#### Expected Promise response
+1. Resolve with a decorated url object for the repository
+2. Reject if not able to get decorate url
+
+### decorateCommit
+Required parameters:
+
+| Parameter        | Type  |  Description |
+| :-------------   | :---- | :-------------|
+| config        | Object | Configuration Object |
+| config.sha     | String | Commit sha to decorate |
+| config.scmUri        | String | Scm uri (ex: `github.com:1234:branchName`) |
+| config.token | String | The github token to check permissions on |
+
+#### Expected Outcome
+Decorated commit in the form of:
+```js
+{
+    url: 'https://github.com/screwdriver-cd/scm-base/commit/5c3b2cc64ee4bdab73e44c394ad1f92208441411',
+    message: 'Use screwdriver to publish',
+    username: 'd2lam'
```

Comment:
Wouldn't this be a whole author instead of just username? According to data-schema 
```
  author:
 +    url: 'https://github.com/d2lam',
 +    name: 'Dao Lam',
 +    username: 'd2lam',
 +    avatar: 'https://avatars3.githubusercontent.com/u/3401924?v=3&s=400'
```